### PR TITLE
[WIP] Add Index and Constraint Names to PSL

### PR DESCRIPTION
**Pull request by do4gr**

### CODE REVIEW

File: introspection-engine/connectors/sql-introspection-connector/src/introspection_helpers.rs [hunk 1]
```diff
@@ -120,36 +121,50 @@ pub fn calculate_many_to_many_field(
...
     };
 
     let default_value = calculate_default(table, &column, &arity);
+    let table_name = table.name.clone();
 
-    let is_unique = table.is_column_unique(&column.name) && !is_id;
+    let is_unique = table
+        .indices
+        .iter()
+        .find(|index| index.tpe == IndexType::Unique && index.columns == vec![column.name.to_string()])
```
Comment (do4gr) [inline introspection-engine/connectors/sql-introspection-connector/src/introspection_helpers.rs:166]: use a slice here

File: libs/datamodel/core/src/transform/attributes/unique_and_index.rs [hunk 2]
```diff
@@ -120,35 +151,62 @@ trait IndexAttributeBase<T>: AttributeValidator<T> {
...
         };
 
-        let fields = args
-            .default_arg("fields")?
-            .as_array()
-            .iter()
-            .map(|f| f.as_constant_literal())
-            .collect::<Result<Vec<_>, _>>()?;
-        index_def.fields = fields;
+        static RE: Lazy<Regex> = Lazy::new(|| Regex::new("[^_a-zA-Z0-9]").unwrap());
```
Comment (do4gr) [inline libs/datamodel/core/src/transform/attributes/unique_and_index.rs:191]: add regex description

File: libs/datamodel/core/tests/attributes/id_negative.rs [hunk 3]
```diff
@@ -139,3 +139,145 @@ fn stringified_field_names_in_id_return_nice_error() {
...
+        name           String            
+        identification Int
+
+        @@id([name, identification], map: "Valid.But.Unwanted")
+    }
+    "#;
+
+    let errors = parse_error(dml);
+    errors.assert_is(DatamodelError::new_model_validation_error(
+        "You defined a database name for the primary key on the model. This is not supported by the provider.",
```
Comment (do4gr) [inline libs/datamodel/core/tests/attributes/id_negative.rs:223]: spell out the provider name

Comment (do4gr) [reply libs/datamodel/core/tests/attributes/id_negative.rs:223]: I'll do that once we have the default provider in place

**Comment by tomhoule** [comment]
This is going to be very, very hard to merge. We should probably start clean and port changes instead.

**Comment by do4gr** [comment]
Closing this as the integration happened via another branch.

**Pull request closed without merging**